packages/vector_graphics_compiler/lib/src/svg/parser.dart

```diff
@@ -113,6 +117,72 @@ class _Elements {
     return null;
   }
 
+  static Future<void>? pattern(SvgParser parserState, bool warningsAsErrors) {
+    final SvgAttributes attributes = parserState._currentAttributes;
+    final String rawWidth = parserState.attribute('width') ?? '';
+    final String rawHeight = parserState.attribute('height') ?? '';
+
+    double? patternWidth;
+    double? patternHeight;
+
+    if (rawWidth.contains('%')) {
+      patternWidth =
+          ((double.parse(rawWidth.substring(0, rawWidth.length - 1))) / 100) *
+              parserState._root!.width;
+    } else if (rawWidth.contains('.') && rawWidth.startsWith('0.')) {
+      patternWidth = (double.parse(rawWidth)) * parserState._root!.width;
+    } else if (rawWidth.isNotEmpty) {
+      patternWidth = (double.parse(rawWidth));
+    }
```

Comment:
Pull this into a helper method in parsers? You might need to return an object that indicates the units, but that might be a better approach than we should follow elsewhere

Reply:
fixed.